### PR TITLE
Support entity query execute dynamic return type whether it is a count query or not

### PR DESCRIPTION
**Pull request by brambaud**
`\Drupal\Core\Entity\Query\QueryInterface::execute` return type depends whether it is a count query or not.

```
/**
   * Execute the query.
   *
   * @return int|array
   *   Returns an integer for count queries or an array of ids. The values of
   *   the array are always entity ids. The keys will be revision ids if the
   *   entity supports revision and entity ids if not.
   */
  public function execute();
```

For the moment only chaining the methods is supported, i.e

```php
\Drupal::entityTypeManager()->getStorage('node')
        ->getQuery()
        ->accessCheck(TRUE)
        ->execute()
```

 I hope to find a way to support cases like another day:

```php
$query = \Drupal::entityTypeManager()->getStorage('node')
    ->getQuery()
    ->accessCheck(TRUE);
$query->execute();
```

### CODE REVIEW

**Comment by mglaman** [comment]
While looking at this, I feel like part of the trick is to also have a dynamic return type on Entity\QueryInterface::count which flags _something_ so we're aware that the parent object type (the query) is a count

**Comment by mglaman** [comment]
@brambaud can you review my push? I ended up rewriting the dynamic return type extension. I added a new custom object type. When we encounter a `count` method, we alter the return type to be this new custom object type. It allows us to check the type and know if it was a count query. Like if the method returned a different interface.

**Comment by brambaud** [comment]
@mglaman I'll try to work on phpstan-drupal at least Friday or this weekend so I'll check that at this moment!
Nice idea for the custom type! I had it too but for #239 😛

**Comment by brambaud** [comment]
Looks great!

It's true `\Drupal::entityQuery` exists too! I'm so used to dependency injection that I forget these static methods exist! 😅 

About the todo, should we wait #239 to be finished and work on this to avoid false positive for config entities?

File: tests/src/Type/EntityQueryDynamicReturnTypeExtensionTest.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace mglaman\PHPStanDrupal\Tests\Type;
```
Comment (brambaud) [inline tests/src/Type/EntityQueryDynamicReturnTypeExtensionTest.php:5]: Shouldn't the test be in a consistent namespace with `mglaman\PHPStanDrupal\Type\EntityQuery\EntityQueryDynamicReturnTypeExtension`?

Something like `mglaman\PHPStanDrupal\Tests\Type\EntityQuery`?

Comment (mglaman) [reply tests/src/Type/EntityQueryDynamicReturnTypeExtensionTest.php]: Yeah we can move it there

**Comment by mglaman** [comment]
@brambaud I think we can handle the config entity storage as a follow up. Technically it's not a regression either way. So we could merge this or that and then address config entity oddities later

File: tests/src/Type/EntityQueryDynamicReturnTypeExtensionTest.php (hunk 1 above)
Comment (brambaud) [reply tests/src/Type/EntityQueryDynamicReturnTypeExtensionTest.php]: Done!

**Comment by brambaud** [comment]
@mglaman Ok! I'll handle the follow up then :)

**Review by mglaman — APPROVED** [review]

**Pull request merged**